blockchain/stake/treasury_test.go

```diff
@@ -0,0 +1,1740 @@
+		0x90, 0xd3, 0x08, 0x4e, 0x0e, 0x31, 0xc7, 0x94,
+		0x22, 0x29, 0x15, 0x3c, 0xe3, 0x4d, 0x87, 0x39,
+		0x29, 0xb1, 0x60, 0x88, 0xd9, 0xe1, 0xaf, 0x43,
+		0x21, // OP_DATA_33 valid public key
+		0x02, 0xa4, 0xf6, 0x45, 0x86, 0xe1, 0x72, 0xc3,
+		0xd9, 0xa2, 0x0c, 0xfa, 0x6c, 0x7a, 0xc8, 0xfb,
+		0x12, 0xf0, 0x11, 0x5b, 0x3f, 0x69, 0xc3, 0xc3,
+		0x5a, 0xec, 0x93, 0x3a, 0x4c, 0x47, 0xc7, 0xd9,
+		0x2c, // No OP_TSPEND
+	}
+
+	// OP_DATA_64 <signature> <pikey> OP_TSPEND OP_TSPEND
+	tspendTwoTSpend []byte = []byte{
+		0x40, // OP_DATA_64 valid signature
+		0x77, 0x69, 0x84, 0xf6, 0x83, 0x13, 0xb1, 0xac,
+		0x62, 0x9e, 0x62, 0x4a, 0xf0, 0x59, 0x5b, 0xdc,
+		0x09, 0xd8, 0xde, 0xd0, 0x2b, 0xc2, 0xb2, 0x9f,
+		0xbd, 0xb3, 0x95, 0x95, 0xe0, 0x3a, 0xc8, 0xb0,
+		0xcf, 0x81, 0x8c, 0xa5, 0x36, 0x72, 0x3e, 0x63,
+		0x90, 0xd3, 0x08, 0x4e, 0x0e, 0x31, 0xc7, 0x94,
+		0x22, 0x29, 0x15, 0x3c, 0xe3, 0x4d, 0x87, 0x39,
+		0x29, 0xb1, 0x60, 0x88, 0xd9, 0xe1, 0xaf, 0x43,
+		0x21, // OP_DATA_33 valid public key
+		0x02, 0xa4, 0xf6, 0x45, 0x86, 0xe1, 0x72, 0xc3,
+		0xd9, 0xa2, 0x0c, 0xfa, 0x6c, 0x7a, 0xc8, 0xfb,
+		0x12, 0xf0, 0x11, 0x5b, 0x3f, 0x69, 0xc3, 0xc3,
+		0x5a, 0xec, 0x93, 0x3a, 0x4c, 0x47, 0xc7, 0xd9,
+		0x2c, // No OP_TSPEND
+		0xc2, // OP_TSPEND
+		0xc2, // Extra OP_TSPEND
+	}
+
+	// OP_DATA_64 <signature> <pikey> OP_TSPEND OP_DATA_1
+	tspendTrailingData []byte = []byte{
+		0x40, // OP_DATA_64 valid signature
+		0x77, 0x69, 0x84, 0xf6, 0x83, 0x13, 0xb1, 0xac,
+		0x62, 0x9e, 0x62, 0x4a, 0xf0, 0x59, 0x5b, 0xdc,
+		0x09, 0xd8, 0xde, 0xd0, 0x2b, 0xc2, 0xb2, 0x9f,
+		0xbd, 0xb3, 0x95, 0x95, 0xe0, 0x3a, 0xc8, 0xb0,
+		0xcf, 0x81, 0x8c, 0xa5, 0x36, 0x72, 0x3e, 0x63,
+		0x90, 0xd3, 0x08, 0x4e, 0x0e, 0x31, 0xc7, 0x94,
+		0x22, 0x29, 0x15, 0x3c, 0xe3, 0x4d, 0x87, 0x39,
+		0x29, 0xb1, 0x60, 0x88, 0xd9, 0xe1, 0xaf, 0x43,
+		0x21, // OP_DATA_33 valid public key
+		0x02, 0xa4, 0xf6, 0x45, 0x86, 0xe1, 0x72, 0xc3,
+		0xd9, 0xa2, 0x0c, 0xfa, 0x6c, 0x7a, 0xc8, 0xfb,
+		0x12, 0xf0, 0x11, 0x5b, 0x3f, 0x69, 0xc3, 0xc3,
+		0x5a, 0xec, 0x93, 0x3a, 0x4c, 0x47, 0xc7, 0xd9,
+		0x2c, // No OP_TSPEND
+		0xc2, // OP_TSPEND
+		0x01, // OP_DATA_1, ByteIndex test in CheckTSpend
+	}
+)
+
+// generateKeys generates all the keys that are hard coded in this file.
+//func generateKeys() {
+//	key := secp256k1.PrivKeyFromBytes(privateKey)
+//	pubKey := key.PubKey()
+//	message := "test message"
+//	messageHash := chainhash.HashB([]byte(message))
+//	signature, err := schnorr.Sign(key, messageHash)
+//	if err != nil {
+//		panic(err)
+//	}
+//	fmt.Printf("Sig 0x%x: %x\n", len(signature.Serialize()),
+//		signature.Serialize())
+//	fmt.Printf("Public key 0x%x: %x\n", len(pubKey.SerializeCompressed()),
+//		pubKey.SerializeCompressed())
+//	for k, v := range signature.Serialize() {
+//		if k%8 == 0 {
+//			fmt.Printf("\n")
+//		}
+//		fmt.Printf("0x%02x,", v)
+//	}
+//	fmt.Printf("\n")
+//}
+//
+//func init() {
+//	generateKeys()
+//	panic("x")
+//}
```

Comment:
I left that code there so that I didn't need to spend half an hour figuring it out again. It proved useful several times and I would prefer to keep it. Even if parameters change it is a valuable hint on how the test values were generated.

Reply:
Just as an FYI, I plan to follow up with PRs removing things like this.  I really, really, really dislike leaving commented junk in the repository for exactly the reason Joe mentioned.  It _always_ gets out of date and results in broken code and it also just clutters things up taking up valuable readers time on things that just aren't needed.  This code is super trivial to write.